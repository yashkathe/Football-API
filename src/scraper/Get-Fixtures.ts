// import axios from "axios";
// import cheerio from "cheerio";

// import { competitionCode } from "../endpoints/variable-codes";
// import { endpoints } from "../endpoints/endpoints";
// import type { isFixture } from "../types/types";

// export const getFixtures = (): Promise<isFixture[]> => {
//     try {
//         const endpoint: string = `${endpoints.competetion}/${competitionCode.premierLeague}/fixtures`;
//         let fixtures: isFixture[] = [];

//         return new Promise(async (resolve, reject) => {
//             const response = await axios(endpoint);
//             const $ = cheerio.load(response.data);

//             $("article.simple-match-card").each(function () {
//                 const homeTeam: string = $(this)
//                     .find("div > div.simple-match-card__teams-content > of-simple-match-card-team:nth-child(1) > div > span.title-8-medium.simple-match-card-team__name")
//                     .text();
//                 console.log(homeTeam);
//             });
//         });
//     } catch (error: any) {
//         throw new Error(error);
//     }
// };
