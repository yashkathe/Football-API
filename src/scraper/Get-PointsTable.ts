import axios from "axios";
import cheerio from "cheerio";

import { competitionCode } from "../endpoints/variable-codes";
import { endpoints } from "../endpoints/endpoints";
import type { isTeamInPointsTable } from "../types/types";

const getPointsTable = (competitionCode: string): Promise<isTeamInPointsTable[]> => {
    try {
        const endpoint: string = `${endpoints.competetion}/${competitionCode}/table`;
        let pointsTable: isTeamInPointsTable[] = [];
        //
        return new Promise(async (resolve, reject) => {
            const response = await axios(endpoint);
            const $ = cheerio.load(response.data);
            //
            $("li.standings__row").each(function () {
                const position: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(1) > span:nth-child(1)").text());
                const team: string = $(this).find("a:nth-child(1) > div:nth-child(3) > p:nth-child(2)").text().trim();
                const gamesPlayed: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(4) > span:nth-child(1)").text());
                const gamesWon: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(5) > span:nth-child(1)").text());
                const gamesLost: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(7) > span:nth-child(1)").text());
                const gamesDrawn: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(6) > span:nth-child(1)").text());
                const goalDifference: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(8) > span:nth-child(1)").text());
                const points: number = parseInt($(this).find("a:nth-child(1) > div:nth-child(9) > span:nth-child(1)").text());

                if (
                    !Number.isNaN(position) &&
                    team.length !== 0 &&
                    !Number.isNaN(gamesPlayed) &&
                    !Number.isNaN(gamesWon) &&
                    !Number.isNaN(gamesLost) &&
                    !Number.isNaN(gamesDrawn) &&
                    !Number.isNaN(goalDifference) &&
                    !Number.isNaN(points)
                ) {
                    const teamInPointsTable: isTeamInPointsTable = {
                        position,
                        team,
                        gamesPlayed,
                        gamesWon,
                        gamesLost,
                        gamesDrawn,
                        goalDifference,
                        points,
                    };
                    pointsTable.push(teamInPointsTable);
                }
            });
            resolve(pointsTable);
        });
    } catch (error: any) {
        throw new Error(error);
    }
};

export const premierLeagueStandings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.premierLeague);
};

export const laLigaStandings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.laliga);
};

export const serieAStandings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.serieA);
};

export const ligue1Standings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.ligue1);
};

export const bundesligaStandings = async (): Promise<isTeamInPointsTable[]> => {
    return await getPointsTable(competitionCode.bundesliga);
};
