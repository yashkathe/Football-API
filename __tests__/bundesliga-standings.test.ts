import { bundesligaStandings } from "../src/scraper/Get-PointsTable";

test("bundesliga standings", async () => {
    expect(await bundesligaStandings()).toMatchSnapshot();
});
