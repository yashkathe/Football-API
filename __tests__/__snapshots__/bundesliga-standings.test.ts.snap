// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bundesliga standings 1`] = `
[
  {
    "gamesDrawn": 4,
    "gamesLost": 1,
    "gamesPlayed": 15,
    "gamesWon": 10,
    "goalDifference": 36,
    "points": 34,
    "position": 1,
    "team": "Bayern Munich",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 3,
    "gamesPlayed": 15,
    "gamesWon": 9,
    "goalDifference": 8,
    "points": 30,
    "position": 2,
    "team": "SC Freiburg",
  },
  {
    "gamesDrawn": 4,
    "gamesLost": 3,
    "gamesPlayed": 15,
    "gamesWon": 8,
    "goalDifference": 9,
    "points": 28,
    "position": 3,
    "team": "RB Leipzig",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 4,
    "gamesPlayed": 15,
    "gamesWon": 8,
    "goalDifference": 8,
    "points": 27,
    "position": 4,
    "team": "Eintracht Frankfurt",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 4,
    "gamesPlayed": 15,
    "gamesWon": 8,
    "goalDifference": 4,
    "points": 27,
    "position": 5,
    "team": "1. FC Union Berlin",
  },
  {
    "gamesDrawn": 1,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 8,
    "goalDifference": 4,
    "points": 25,
    "position": 6,
    "team": "Borussia Dortmund",
  },
  {
    "gamesDrawn": 5,
    "gamesLost": 4,
    "gamesPlayed": 15,
    "gamesWon": 6,
    "goalDifference": 4,
    "points": 23,
    "position": 7,
    "team": "VfL Wolfsburg",
  },
  {
    "gamesDrawn": 4,
    "gamesLost": 5,
    "gamesPlayed": 15,
    "gamesWon": 6,
    "goalDifference": 4,
    "points": 22,
    "position": 8,
    "team": "M'gladbach",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 6,
    "goalDifference": -2,
    "points": 21,
    "position": 9,
    "team": "Werder Bremen",
  },
  {
    "gamesDrawn": 4,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 5,
    "goalDifference": -5,
    "points": 19,
    "position": 10,
    "team": "Mainz 05",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 7,
    "gamesPlayed": 15,
    "gamesWon": 5,
    "goalDifference": 0,
    "points": 18,
    "position": 11,
    "team": "Hoffenheim",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 7,
    "gamesPlayed": 15,
    "gamesWon": 5,
    "goalDifference": -1,
    "points": 18,
    "position": 12,
    "team": "Bayer Leverkusen",
  },
  {
    "gamesDrawn": 5,
    "gamesLost": 6,
    "gamesPlayed": 15,
    "gamesWon": 4,
    "goalDifference": -8,
    "points": 17,
    "position": 13,
    "team": "1. FC KÃ¶ln",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 8,
    "gamesPlayed": 15,
    "gamesWon": 4,
    "goalDifference": -8,
    "points": 15,
    "position": 14,
    "team": "Augsburg",
  },
  {
    "gamesDrawn": 5,
    "gamesLost": 7,
    "gamesPlayed": 15,
    "gamesWon": 3,
    "goalDifference": -3,
    "points": 14,
    "position": 15,
    "team": "Hertha BSC",
  },
  {
    "gamesDrawn": 5,
    "gamesLost": 7,
    "gamesPlayed": 15,
    "gamesWon": 3,
    "goalDifference": -9,
    "points": 14,
    "position": 16,
    "team": "VfB Stuttgart",
  },
  {
    "gamesDrawn": 1,
    "gamesLost": 10,
    "gamesPlayed": 15,
    "gamesWon": 4,
    "goalDifference": -22,
    "points": 13,
    "position": 17,
    "team": "VfL Bochum",
  },
  {
    "gamesDrawn": 3,
    "gamesLost": 10,
    "gamesPlayed": 15,
    "gamesWon": 2,
    "goalDifference": -19,
    "points": 9,
    "position": 18,
    "team": "Schalke 04",
  },
]
`;
